### PR TITLE
Dyn 5245 Dynamo-Revit notifications_center crashing and not loading notifications

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose
Load notifications in Dynamo for Revit and avoid crashing when exiting Dynamo (when notifications being displayed).
I added code for create the WebView2 cache folder in the AppData location because when tries to create it inside C:\Program Files\Autodesk\Revit 2023\AddIns\DynamoForRevit is crashing due to permissions.
Also I added some code for stopping the WebView2 instance when Dynamo is being closed because otherwise is crashing.
Also the Dispose method was deleted due that we are doing the same in the new method added SuspendCoreWebviewAsync.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Load notifications in Dynamo for Revit and avoid crashing when exiting Dynamo (when notifications being displayed).

### Reviewers

@QilongTang 

### FYIs

@filipeotero @jesusalvino

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
GIF showing Dynamo on Revit is not crashing and showing notifications.
DynamoRevit_NoCrash

File: src/Notifications/NotificationCenterController.cs [hunk 1]
```diff
@@ -58,11 +60,16 @@ internal NotificationCenterController(DynamoView view, DynamoLogger dynLogger)
         {
             dynamoView = view;
             dynamoViewModel = dynamoView.DataContext as DynamoViewModel;
+            //When executing Dynamo as Sandbox or inside any host like Revit, FormIt, Civil3D the WebView2 cache folder will be located in the AppData folder
+            var userDataDir = new DirectoryInfo(dynamoViewModel.Model.PathManager.UserDataDirectory);
+            webBrowserUserDataFolder = userDataDir.Exists ? userDataDir : null;
```
Comment (QilongTang) [inline src/Notifications/NotificationCenterController.cs:65]: So this is actually the step missing before, causing the browser to be empty?

**Review by QilongTang — APPROVED** [review]

File: src/Notifications/NotificationCenterController.cs [hunk 2]
```diff
@@ -72,17 +79,56 @@ internal NotificationCenterController(DynamoView view, DynamoLogger dynLogger)
...
+        }
+
+        async void SuspendCoreWebviewAsync()
+        {
+            notificationUIPopup.IsOpen = false;
+            notificationUIPopup.webView.Visibility = Visibility.Hidden;
+
+            if (notificationUIPopup.webView.CoreWebView2 != null)
+            {
+                await notificationUIPopup.webView.CoreWebView2.TrySuspendAsync();
```
Comment (mjkkirschner) [inline src/Notifications/NotificationCenterController.cs:105]: why even do this, it appears to sleep the webview, when we call stop on the next line?

**Review by reddyashish — APPROVED** [review]
LGTM after addressing Mike's comment

File: src/Notifications/NotificationCenterController.cs [hunk 3]
```diff
@@ -72,17 +79,56 @@ internal NotificationCenterController(DynamoView view, DynamoLogger dynLogger)
...
             };
-            notificationUIPopup.webView.EnsureCoreWebView2Async();
-            notificationUIPopup.webView.CoreWebView2InitializationCompleted += WebView_CoreWebView2InitializationCompleted;
             logger = dynLogger;
             // If user turns on the feature, they will need to restart Dynamo to see the count
             // This ensures no network traffic when Notification center feature is turned off
             if (dynamoViewModel.PreferenceSettings.EnableNotificationCenter) 
             {
                 RequestNotifications();
+                InitializeBrowserAsync();
```
Comment (QilongTang) [inline src/Notifications/NotificationCenterController.cs:88]: @RobertGlobant20 @mjkkirschner I think this line is not meant to be here? With this, the constructor will always try to initialize WebView2

**Comment by RobertGlobant20** [comment]
The code for creating the UserDataFolder and for preventing the Crash will be divided in two pull request so we can address comments in CI jobs.

**Pull request closed without merging**